docs/EC2.md

```diff
@@ -3,9 +3,43 @@
 Vault Control Tool can run both on startup to authenticate an EC2 instance to Vault, as well as periodically
 in a cronjob to keep various secrets fresh for your system.
 
-Note that EC2 support is new, please file issues as you encounter them.
+Vault Control Tool supports both authentication types present in the aws auth method - iam and ec2.
 
-## Setup
+## IAM
+
+### Setup
+
+Assuming you have already bound an IAM role to your EC2 instance, you'll need to create a role in Vault that associates 
+this IAM role with specific policies.
+
+```bash
+vault write auth/aws/role/example-role auth_type=iam bound_iam_principal_arn=example-iam-role-arn policies=example-policy max_ttl=500h
+```
+
+Vault Control Tool requires the `--iam-auth-role` flag to be set to this role name in order to authenticate to Vault using it.
+
+### On Startup
+
+Assuming you have a `vault-config.yml` in `/etc/vault-ctrl-tool`, initialization is as easy as:
+
+```bash
+export VAULT_ADDR=https://vault.service.consul:8200/
+vault-ctrl-tool --iam-auth-role=example-role --init --input-prefix=/etc/vault-ctrl-tool --output-prefix=/etc/vault-ctrl-tool
+```
+
+### Sidecar
+
+You can either launch Vault Control Tool as a process, or call it from cron. The later is the recommended
+mechanism:
+
+```bash
+export VAULT_ADDR=https://vault.service.consul:8200/
+vault-ctrl-tool --iam-auth-role=example-role --sidecar --one-shot --input-prefix=/etc/vault-ctrl-tool --output-prefix=/etc/vault-ctrl-tool
```

Comment:
Please bake the iam-auth-role into the lease so that the sidecar can run without needing it specified.